clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -972,16 +975,28 @@ private Record nextFetchedRecord() {
         private List<ConsumerRecord<K, V>> fetchRecords(int maxRecords) {
             if (isFetched)
                 return Collections.emptyList();
+            if (nextInlineException != null) {
```

Comment:
Yes, this behavior may have been unintentionally lost when I implemented the streaming parsing. That means we are missing a test case to validate the behavior (see my other comment). Note I am not saying this behavior is definitely the way forward, but that this patch is not a good place to change it.